addon/components/select-2.js

```diff
@@ -505,6 +590,9 @@ var Select2Component = Ember.Component.extend({
       this._select.select2("val", value);
     } else {
       // otherwise set the full object via "data"
+      if (value instanceof Ember.ArrayProxy) {
```

Comment:
I had to change `Ember.ArrayProxy` to `DS.ManyArray` to get this to work as expected. I'm on Ember 1.13.7 and Ember Data 1.13.8, and when I use the `hasMany` relationship as `value`, this is false, and my values aren't properly displayed. Obviously this would break for those not using Ember Data.